### ISSUE TITLE
Configuring TLS?

### ISSUE THREAD

**Issue by dsully** [kurtbuilds/httpclient#5]
Hi - I need to modify the TLS configuration (specifically adding a CA bundle, and in some cases using an insecure connection).

However, TLS is setup as so:

```rust
fn https_connector() -> &'static HttpsConnector<HttpConnector> {
    HTTPS_CONNECTOR.get_or_init(|| {
        hyper_rustls::HttpsConnectorBuilder::new()
            .with_native_roots()
            .https_or_http()
            .enable_http1()
            .build()
    })
}
```

Would it be possible to update this to take in a connector?

Thanks!

**Comment by kurtbuilds** [comment]
Definitely. My week is slammed, but a PR is welcome.

**Comment by dsully** [comment]
https://github.com/kurtbuilds/httpclient/pull/6

I'd also like to set dependency versions at floating major if that's ok.

**Comment by kurtbuilds** [comment]
Thank you, looks good to me.

What deps do you want float on?

**Comment by dsully** [comment]
```
cookie = { version = "0.18.0", features = [ "percent-encode" ] }
encoding_rs = "0.8.30"
futures = "0.3.25"
indexmap = "2.1.0"
rand = "0.8.5"
regex = "1.7.1"
serde = { version = "1.0.136", features = [ "derive" ] }
serde_json = "1.0.79"
serde_qs = "0.12.0"
tracing = "0.1.37"
urlencoding = "2.1.0"
walkdir = "2.3.2"
```

I believe all of those could/should be floated. `tokio` as well.

**Comment by kurtbuilds** [comment]
Merged #6 in.

**Comment by kurtbuilds** [comment]
The floating thing doesn't make much sense to me. If everyone appropriately respects semver, if they have breaking API changes, they'll break this, and non-breaking changes will be included appropriately. Happy to discuss more.
